relay-general/src/store/normalize.rs

```diff
@@ -3010,4 +3053,67 @@ mod tests {
         )
         "###);
     }
+
+    #[test]
+    fn test_keeps_valid_measurement() {
+        let name = "lcp";
+        let measurement = Measurement {
+            value: Annotated::new(420.69),
+            unit: Annotated::new(MetricUnit::Duration(DurationUnit::MilliSecond)),
+        };
+        let should_keep = true;
+
+        test_invalid_measurement(name, measurement, should_keep);
+    }
+
+    #[test]
+    fn test_drops_too_long_measurement_names() {
+        let name = "lcpppppppppppppppppppppppppppp";
+        let measurement = Measurement {
+            value: Annotated::new(420.69),
+            unit: Annotated::new(MetricUnit::Duration(DurationUnit::MilliSecond)),
+        };
+        let should_keep = false;
+
+        test_invalid_measurement(name, measurement, should_keep);
+    }
+
+    #[test]
+    fn test_drops_measurements_with_invalid_characters() {
+        let name = "i æm frøm nørwåy";
+        let measurement = Measurement {
+            value: Annotated::new(420.69),
+            unit: Annotated::new(MetricUnit::Duration(DurationUnit::MilliSecond)),
+        };
+        let should_keep = false;
+
+        test_invalid_measurement(name, measurement, should_keep);
+    }
+
+    fn test_invalid_measurement(name: &str, measurement: Measurement, should_keep: bool) {
```

Comment:
This approach is not bad, but I think it's harder to understand than building a payload and passing it directly, such as [here](https://github.com/getsentry/relay/blob/b597efdbabaf47fae561e0a6b4e5aabe4dade683/relay-general/src/store/normalize.rs#L1952).

If you feel like trying something new and want to play around a bit with a macro, see how we do with transaction names: [this](https://github.com/getsentry/relay/blob/fcbd996ace227a9fe9c69b736db91482e5e178d5/relay-general/src/store/transactions/processor.rs#L1996) macro, and its usage [here](https://github.com/getsentry/relay/blob/fcbd996ace227a9fe9c69b736db91482e5e178d5/relay-general/src/store/transactions/processor.rs#LL2037C28-L2037C28).